src/backend/gencomm.py

```diff
@@ -93,7 +96,14 @@ def switcher_builtin(backend, OUTFILE, blklens, builtin_types, builtin_maps, typ
     yutils.display(OUTFILE, "switch (%s->u.builtin.handle) {\n" % child_type_str(typelist))
 
     for b in builtin_types:
-        switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE__%s" % b.replace(" ", "_"), b.replace(" ", "_"))
+        if (b == "c_complex"):
+            switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE__C_COMPLEX", b.replace(" ", "_"))
+        elif (b == "c_double_complex"):
+            switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE__C_DOUBLE_COMPLEX", b.replace(" ", "_"))
+        elif (b == "c_long_double_complex"):
+            switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE__C_LONG_DOUBLE_COMPLEX", b.replace(" ", "_"))
+        else:
+            switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE__%s" % b.replace(" ", "_"), b.replace(" ", "_"))
```

Comment:
Why do we need separate cases? It appears the `else` branch will work for the complex names, right?

Reply:
Agreed. I will fix it.